runtime/ztests/expr/function/grok.yaml

```diff
@@ -0,0 +1,60 @@
+zed: yield grok(pattern, input, defs)
+
+vector: true
+
+input: |
+  {
+    pattern: "%{TIMESTAMP_ISO8601:event_time} %{LOGLEVEL:log_level} %{GREEDYDATA:log_message}",
+    input: "2020-09-16T04:20:42.45+01:00 DEBUG This is a sample debug log message",
+    defs: null(string)
+  }
```

Comment:
While you're here, maybe add test cases for null(string) pattern and input along with non-string pattern, input, and defs?

Reply:
What do you think they should do with null values? Is the current implementation incorrect?

Reply:
I'll just have it return a null empty record when null inputs are encountered. Seems consistent.